app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -239,4 +253,27 @@ private void importDatabase(final String filePath) {
             ErrorActivity.reportUiErrorInSnackbar(this, "Importing database", e);
         }
     }
+
+    private boolean isValidPath(final String path) {
+        if (path == null || path.isEmpty()) {
+            return false;
+        }
+        final File file = new File(path);
+        return file.exists() && file.isDirectory();
+    }
+
+    private void setImportExportDataPath(final File file) {
+        final String directoryPath;
+        if (!file.isDirectory()) {
+            final File parentFile = file.getParentFile();
+            if (parentFile != null) {
+                directoryPath = parentFile.getAbsolutePath();
+            } else {
+                directoryPath = "";
+            }
+        } else {
+            directoryPath = file.getAbsolutePath();
+        }
```

Comment:
I feel like this would be easier to read, if the `!file.isDirectory()` is flipped to `file.isDirectory()` and the bodies are switched accordingly

Reply:
fixed